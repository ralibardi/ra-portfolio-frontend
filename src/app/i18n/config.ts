import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import HttpApi from 'i18next-http-backend';

i18n
  .use(LanguageDetector)
  .use(HttpApi)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en-GB',
    debug: process.env.NODE_ENV === 'development',
    interpolation: {
      escapeValue: false,
    },
    detection: {
      order: ['navigator', 'htmlTag', 'path', 'subdomain'],
      caches: ['localStorage', 'cookie'],
    },
    backend: {
      loadPath:
        process.env.NODE_ENV === 'production'
          ? `${window.location.origin}/locales/{{lng}}/{{ns}}.json`
          : '/locales/{{lng}}/{{ns}}.json',
      requestOptions: {
        cache: 'no-store',
      },
    },
    load: 'languageOnly',
  })
  .catch(console.error);

export default i18n;
