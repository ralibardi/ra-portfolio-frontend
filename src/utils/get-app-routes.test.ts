import { getAppRoutes } from './get-app-routes';
import {
  homePagePath,
  codingPagePath,
  photographyPagePath,
  gamingPagePath,
  aboutPagePath,
  contactPagePath,
  cvPagePath,
  healthPagePath,
  errorPagePath,
} from './route-paths';
import { PAGES } from '@app/i18n/keys';
import {
  faHome,
  faTerminal,
  faCamera,
  faGamepad,
  faUser,
  faAddressCard,
  faFile,
  faStethoscope,
  faX,
} from '@fortawesome/free-solid-svg-icons';

describe('getAppRoutes', () => {
  it('should export an array with 9 routes', () => {
    expect(Array.isArray(getAppRoutes)).toBe(true);
    expect(getAppRoutes).toHaveLength(9);
  });

  it('should have correct properties for the home route', () => {
    const homeRoute = getAppRoutes.find((route) => route.path === homePagePath);
    expect(homeRoute).toBeDefined();
    expect(homeRoute?.index).toBe(true);
    expect(homeRoute?.labelKey).toBe(PAGES.HOME.NAME);
    expect(homeRoute?.icon).toEqual(faHome);
    expect(homeRoute?.enabled).toBe(true);
    expect(homeRoute?.hidden).toBe(false);
    expect(homeRoute?.component).toBeDefined();
  });

  it('should have correct properties for the coding route', () => {
    const codingRoute = getAppRoutes.find(
      (route) => route.path === codingPagePath,
    );
    expect(codingRoute).toBeDefined();
    expect(codingRoute?.index).toBe(true);
    expect(codingRoute?.labelKey).toBe(PAGES.CODING.NAME);
    expect(codingRoute?.icon).toEqual(faTerminal);
    expect(codingRoute?.enabled).toBe(false);
    expect(codingRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the photography route', () => {
    const photographyRoute = getAppRoutes.find(
      (route) => route.path === photographyPagePath,
    );
    expect(photographyRoute).toBeDefined();
    expect(photographyRoute?.index).toBe(true);
    expect(photographyRoute?.labelKey).toBe(PAGES.PHOTOGRAPHY.NAME);
    expect(photographyRoute?.icon).toEqual(faCamera);
    expect(photographyRoute?.enabled).toBe(false);
    expect(photographyRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the gaming route', () => {
    const gamingRoute = getAppRoutes.find(
      (route) => route.path === gamingPagePath,
    );
    expect(gamingRoute).toBeDefined();
    expect(gamingRoute?.index).toBe(true);
    expect(gamingRoute?.labelKey).toBe(PAGES.GAMING.NAME);
    expect(gamingRoute?.icon).toEqual(faGamepad);
    expect(gamingRoute?.enabled).toBe(false);
    expect(gamingRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the about route', () => {
    const aboutRoute = getAppRoutes.find(
      (route) => route.path === aboutPagePath,
    );
    expect(aboutRoute).toBeDefined();
    expect(aboutRoute?.index).toBe(true);
    expect(aboutRoute?.labelKey).toBe(PAGES.ABOUT.NAME);
    expect(aboutRoute?.icon).toEqual(faUser);
    expect(aboutRoute?.enabled).toBe(false);
    expect(aboutRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the contact route', () => {
    const contactRoute = getAppRoutes.find(
      (route) => route.path === contactPagePath,
    );
    expect(contactRoute).toBeDefined();
    expect(contactRoute?.index).toBe(true);
    expect(contactRoute?.labelKey).toBe(PAGES.CONTACT.NAME);
    expect(contactRoute?.icon).toEqual(faAddressCard);
    expect(contactRoute?.enabled).toBe(false);
    expect(contactRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the cv route', () => {
    const cvRoute = getAppRoutes.find((route) => route.path === cvPagePath);
    expect(cvRoute).toBeDefined();
    expect(cvRoute?.index).toBe(true);
    expect(cvRoute?.labelKey).toBe(PAGES.CV.NAME);
    expect(cvRoute?.icon).toEqual(faFile);
    expect(cvRoute?.enabled).toBe(false);
    expect(cvRoute?.hidden).toBe(false);
  });

  it('should have correct properties for the health route', () => {
    const healthRoute = getAppRoutes.find(
      (route) => route.path === healthPagePath,
    );
    expect(healthRoute).toBeDefined();
    expect(healthRoute?.index).toBe(true);
    expect(healthRoute?.labelKey).toBe(PAGES.HEALTH.NAME);
    expect(healthRoute?.icon).toEqual(faStethoscope);
    expect(healthRoute?.enabled).toBe(true);
    expect(healthRoute?.hidden).toBe(true);
  });

  it('should have correct properties for the error route', () => {
    const errorRoute = getAppRoutes.find(
      (route) => route.path === errorPagePath,
    );
    expect(errorRoute).toBeDefined();
    expect(errorRoute?.index).toBe(true);
    expect(errorRoute?.labelKey).toBe(PAGES.ERROR.NAME);
    expect(errorRoute?.icon).toEqual(faX);
    expect(errorRoute?.enabled).toBe(true);
    expect(errorRoute?.hidden).toBe(true);
  });

  it('should match the snapshot', () => {
    expect(getAppRoutes).toMatchSnapshot();
  });
});
