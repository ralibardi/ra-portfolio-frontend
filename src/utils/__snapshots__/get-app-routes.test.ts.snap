// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAppRoutes should match the snapshot 1`] = `
[
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": true,
    "hidden": false,
    "icon": {
      "icon": [
        576,
        512,
        [
          127968,
          63498,
          63500,
          "home",
          "home-alt",
          "home-lg-alt",
        ],
        "f015",
        "M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z",
      ],
      "iconName": "house",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.home.name",
    "path": "/",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        576,
        512,
        [],
        "f120",
        "M9.4 86.6C-3.1 74.1-3.1 53.9 9.4 41.4s32.8-12.5 45.3 0l192 192c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L178.7 256 9.4 86.6zM256 416l288 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-288 0c-17.7 0-32-14.3-32-32s14.3-32 32-32z",
      ],
      "iconName": "terminal",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.coding.name",
    "path": "/coding",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        512,
        512,
        [
          62258,
          "camera-alt",
        ],
        "f030",
        "M149.1 64.8L138.7 96 64 96C28.7 96 0 124.7 0 160L0 416c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-256c0-35.3-28.7-64-64-64l-74.7 0L362.9 64.8C356.4 45.2 338.1 32 317.4 32L194.6 32c-20.7 0-39 13.2-45.5 32.8zM256 192a96 96 0 1 1 0 192 96 96 0 1 1 0-192z",
      ],
      "iconName": "camera",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.photography.name",
    "path": "/photography",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        640,
        512,
        [],
        "f11b",
        "M192 64C86 64 0 150 0 256S86 448 192 448l256 0c106 0 192-86 192-192s-86-192-192-192L192 64zM496 168a40 40 0 1 1 0 80 40 40 0 1 1 0-80zM392 304a40 40 0 1 1 80 0 40 40 0 1 1 -80 0zM168 200c0-13.3 10.7-24 24-24s24 10.7 24 24l0 32 32 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-32 0 0 32c0 13.3-10.7 24-24 24s-24-10.7-24-24l0-32-32 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l32 0 0-32z",
      ],
      "iconName": "gamepad",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.gaming.name",
    "path": "/gaming",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        448,
        512,
        [
          128100,
          62144,
        ],
        "f007",
        "M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512l388.6 0c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304l-91.4 0z",
      ],
      "iconName": "user",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.about.name",
    "path": "/about",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        576,
        512,
        [
          62140,
          "contact-card",
          "vcard",
        ],
        "f2bb",
        "M64 32C28.7 32 0 60.7 0 96L0 416c0 35.3 28.7 64 64 64l448 0c35.3 0 64-28.7 64-64l0-320c0-35.3-28.7-64-64-64L64 32zm80 256l64 0c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16L80 384c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32l128 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-128 0c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64l128 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-128 0c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64l128 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-128 0c-8.8 0-16-7.2-16-16s7.2-16 16-16z",
      ],
      "iconName": "address-card",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.contact.name",
    "path": "/contact",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": false,
    "hidden": false,
    "icon": {
      "icon": [
        384,
        512,
        [
          128196,
          128459,
          61462,
        ],
        "f15b",
        "M0 64C0 28.7 28.7 0 64 0L224 0l0 128c0 17.7 14.3 32 32 32l128 0 0 288c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 64zm384 64l-128 0L256 0 384 128z",
      ],
      "iconName": "file",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.cv.name",
    "path": "/cv",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": true,
    "hidden": true,
    "icon": {
      "icon": [
        576,
        512,
        [
          129658,
        ],
        "f0f1",
        "M142.4 21.9c5.6 16.8-3.5 34.9-20.2 40.5L96 71.1 96 192c0 53 43 96 96 96s96-43 96-96l0-120.9-26.1-8.7c-16.8-5.6-25.8-23.7-20.2-40.5s23.7-25.8 40.5-20.2l26.1 8.7C334.4 19.1 352 43.5 352 71.1L352 192c0 77.2-54.6 141.6-127.3 156.7C231 404.6 278.4 448 336 448c61.9 0 112-50.1 112-112l0-70.7c-28.3-12.3-48-40.5-48-73.3c0-44.2 35.8-80 80-80s80 35.8 80 80c0 32.8-19.7 61-48 73.3l0 70.7c0 97.2-78.8 176-176 176c-92.9 0-168.9-71.9-175.5-163.1C87.2 334.2 32 269.6 32 192L32 71.1c0-27.5 17.6-52 43.8-60.7l26.1-8.7c16.8-5.6 34.9 3.5 40.5 20.2zM480 224a32 32 0 1 0 0-64 32 32 0 1 0 0 64z",
      ],
      "iconName": "stethoscope",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.health.name",
    "path": "/health",
  },
  {
    "component": {
      "$$typeof": Symbol(react.lazy),
      "_init": [Function],
      "_payload": {
        "_result": [Function],
        "_status": -1,
      },
    },
    "enabled": true,
    "hidden": true,
    "icon": {
      "icon": [
        384,
        512,
        [
          120,
        ],
        "58",
        "M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z",
      ],
      "iconName": "x",
      "prefix": "fas",
    },
    "index": true,
    "labelKey": "pages.error.name",
    "path": "/error",
  },
]
`;
